src/components/checkbox/checkbox.spec.ts

```diff
@@ -0,0 +1,749 @@
+import {
+  beforeEachProviders,
+  fakeAsync,
+  inject,
+  injectAsync,
+  tick,
+  ComponentFixture,
+  TestComponentBuilder,
+} from 'angular2/testing';
+
+import {
+  it,
+  iit,
+  describe,
+  ddescribe,
+  expect,
+  beforeEach,
+} from '../../core/facade/testing';
+
+import {provide, Component, DebugElement, EventEmitter} from 'angular2/core';
+import {By} from 'angular2/platform/browser';
+
+import {MdCheckbox} from './checkbox';
+
+export function main() {
+  describe('MdCheckbox', function() {
+    var builder: TestComponentBuilder;
+
+    beforeEach(inject([TestComponentBuilder], function(tcb: TestComponentBuilder) {
+      builder = tcb;
+    }));
+
+    it('attaches a class "md-checkbox" to the host element', function(done: () => void) {
+      builder.createAsync(CheckboxController).then(function(fixture) {
+        fixture.detectChanges();
+        const el = fixture.debugElement.query(By.css('.md-checkbox'));
+        expect(el).not.toBeNull();
+      }).then(done).catch(done);
+    });
+
+    it('creates a checkbox frame element', function(done: () => void) {
```

Comment:
not sure I 100% agree with that, as I've definitely been bitten in the past by refactoring code and accidentally deleting a char from a className resulting in time spent tracking down where I made that mistake. Just my unsolicited $0.02 though; I'll go ahead and remove them.